src/pybind/mgr/influx/module.py

```diff
@@ -265,70 +342,74 @@ def send_to_influx(self):
                     'detail': ['Configuration option hostname not set']
                 }
             })
-            return
+            return False
+
+        self.set_health_checks(dict())
```

Comment:
This is racing with the worker threads' calls to set_health_checks, I think?  Perhaps its better to have each worker thread advertise its own health issues (subclass Thread, give it an attribute for its health state), and then gather them and do set_health_checks from here.